src/handlers/ motions/motionCreated.ts

```diff
@@ -12,23 +13,30 @@ export default async (event: ContractEvent) => {
     args: { motionId },
   } = event;
 
-  const colonyClient = await networkClient.getColonyClient(colonyAddress);
-  const parsedAction = await getParsedActionFromMotion(motionId, colonyClient);
+  try {
+    const colonyClient = await networkClient.getColonyClient(colonyAddress);
+    const motionData = await extractDataFromMotion(
+      motionId,
+      colonyClient,
+      colonyAddress,
+    );
 
-  if (parsedAction) {
-    const contractOperation = parsedAction.name;
+    const contractOperation = motionData.parsedAction.name;
 
     /* Handle the action type-specific mutation here*/
     switch (contractOperation) {
       case ColonyOperations.MintTokens: {
-        await writeMintTokensMotionToDB(event, parsedAction, colonyClient);
+        await writeMintTokensMotionToDB(event, motionData);
         break;
       }
       default: {
         break;
       }
     }
-
     verbose(`${contractOperation} Motion Created`);
+
+    await motionSpecificEventsListener(colonyAddress);
```

Comment:
I am only listening for motion staked events once a motion has been created. Which is fine for now, so long as you don't restart the ingestor mid-motion in testing (then you'd need to create a new motion).

However, when it comes to tracking motions, this might change.